pkg/helm/actions/install_chart.go

```diff
@@ -1,22 +1,118 @@
 	"helm.sh/helm/v3/pkg/chart"
 	"helm.sh/helm/v3/pkg/chart/loader"
 	"helm.sh/helm/v3/pkg/release"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog"
+)
+
+var (
+	helmChartRepositoryClusterGVK = schema.GroupVersionResource{
+		Group:    "helm.openshift.io",
+		Version:  "v1beta1",
+		Resource: "helmchartrepositories",
+	}
+	helmChartRepositoryNamespaceGVK = schema.GroupVersionResource{
+		Group:    "helm.openshift.io",
+		Version:  "v1beta1",
+		Resource: "projecthelmchartrepositories",
+	}
 )
 
-func InstallChart(ns, name, url string, vals map[string]interface{}, conf *action.Configuration) (*release.Release, error) {
+func InstallChart(ns, name, url string, vals map[string]interface{}, conf *action.Configuration, client dynamic.Interface, coreClient corev1client.CoreV1Interface) (*release.Release, error) {
 	cmd := action.NewInstall(conf)
+	// tlsFiles contain references of files to be removed once the chart
+	// operation depending on those files is finished.
+	tlsFiles := []*os.File{}
+	var tlsConfigNamespace, configMapName, secretName string
+	repositoryName, _, err := getChartNameAndNamespaceFromChartUrl(url, ns, client, coreClient)
+	if err != nil {
+		//serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to parse request: %v", err)})
+		return nil, err
+	}
+	// Create a Kubernetes core/v1 client.
+	connectionConfig, err := getRepoConnectionConfig(repositoryName, ns, client)
+	if err != nil {
+		//serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: fmt.Sprintf("Failed to parse request: %v", err)})
+		return nil, err
+	}
+	if connectionConfig != (&v1beta1.ConnectionConfig{}) {
+		if connectionConfig.TLSClientConfig != nil {
+			secretName = connectionConfig.TLSClientConfig.Name
+			tlsConfigNamespace = connectionConfig.TLSClientConfig.Namespace
+			if tlsConfigNamespace == "" {
+				tlsConfigNamespace = configNamespace
+			}
+		}
+		if connectionConfig.CA != nil {
+			configMapName = connectionConfig.CA.Name
+		}
 
+		if configMapName != "" {
+			configMap, err := coreClient.ConfigMaps(configNamespace).Get(context.TODO(), configMapName, v1.GetOptions{})
+			if err != nil {
+				return nil, errors.New(fmt.Sprintf("Failed to GET configmap %s, reason %v", configMapName, err))
+			}
+			caCertBytes, found := configMap.Data[caBundleKey]
+			if !found {
+				return nil, errors.New(fmt.Sprintf("Failed to find %s key in configmap %s", caBundleKey, configMapName))
+			}
+			caCertFile, caCertGetErr := writeTempFile([]byte(caCertBytes), "cacert-*")
+			if caCertGetErr != nil {
+				return nil, caCertGetErr
+			}
+			cmd.ChartPathOptions.CaFile = caCertFile.Name()
+			tlsFiles = append(tlsFiles, caCertFile)
+		}
+		if secretName != "" {
```

Comment:
This conditional should be moved to around line 56, since `secretName` only receives a value if the conditional of line 55 is true.

Reply:
yeah sounds correct